lib/src/view/design_main_strand.dart

```diff
@@ -189,15 +189,24 @@ class DesignMainStrandComponent extends UiComponent2<DesignMainStrandProps>
     app.dispatch(action);
   }
 
-  assign_domain_name_complement_from_bound_strands() {
-    List<Strand> strands_selected = app.state.ui_state.selectables_store.selected_strands.toList();
+  assign_domain_name_complement_from_bound_strands({Domain domain}) {
+    if (app.state.ui_state.select_mode_state.domains_selectable()) {
+      List<Domain> domains_selected = app.state.ui_state.selectables_store.selected_domains.toList();
 
-    if (!strands_selected.contains(props.strand)) {
-      strands_selected.add(props.strand);
-    }
+      if (!domains_selected.contains(domain)) {
+        domains_selected.add(domain);
+      }
 
-    var action = actions.AssignDomainNameComplementFromBoundStrands(strands_selected);
-    app.dispatch(action);
+      var action = actions.AssignDomainNameComplementFromBoundDomains(domains_selected);
+      app.dispatch(action);
+    } else {
+      List<Strand> strands_selected = app.state.ui_state.selectables_store.selected_strands.toList();
+      if (!strands_selected.contains(props.strand)) {
+        strands_selected.add(props.strand);
+      }
+      var action = actions.AssignDomainNameComplementFromBoundStrands(strands_selected);
+      app.dispatch(action);
+    }
   }
 
   assign_domain_name_complement_from_bound_domains(Domain domain) {
```

Comment:
As far as I can tell, the function `assign_domain_name_complement_from_bound_domains` is not being called anywhere. If that's the case, it should be removed.